core/arch/arm/include/mm/core_mmu.h

```diff
@@ -77,6 +77,11 @@
 #define CFG_TEE_RAM_VA_SIZE		CORE_MMU_PGDIR_SIZE
 #endif
 
+#ifdef CFG_WITH_LPAE
+#define CORE_MMU_L1_TBL_OFFSET		(CFG_TEE_CORE_NB_CORE * 4 * 8)
```

Comment:
` * 4 * 8` looks like magic, especially compared to how `NUM_L1_ENTRIES` is defined in **core_mmu_lpae.c**. Deserves a bit of explanation: ~~why~~ 4 64bit wide mmu descriptors to cover 4x1GB ?

Regarding the armv7, the comment is straightforward: we need a full L1 16kB wide table.

Reply:
I could base this on `NUM_L1_ENTRIES`, but that means moving all needed defines in `core_mmu_lpae.c` to this file which are overlapping with the defines above in this file.
That would warrant a cleanup that I don't think we want to do right now.

How about just adding a comment:
```
/*
 * CORE_MMU_L1_TBL_OFFSET is used when switching to/from reduced kernel mapping.
 * The actual value depends on internals in core_mmu_lpae.c and core_mmu_v7.c which we
 * rather not expose here. There's a compile time assertion to check that these magic numbers
 * are correct.
 */
```